include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -123,38 +125,58 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 
     // TODO: filter hits and only do it when necessary (depending on error types)
 
+    constexpr bool output_cursor = cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>();
+
+    using result_t = std::conditional_t<output_cursor,
+                                        search_result_value_type<typename index_t::cursor_type>,
+                                        search_result_value_type<typename index_t::cursor_type, size_t, size_t>>;
+
+    std::vector<result_t> results;
```

Comment:
Shouldn't this be `std::vector<search_result<search_result_value_t>>`?

Reply:
I'd like to, but we need to enumerate the queries in the caller `search_all()` and hence need to modify the `search_result_value_type`